Source/WebKit/Shared/LogStream.cpp

```diff
@@ -68,13 +69,10 @@ void LogStream::logOnBehalfOfWebContent(std::span<const uint8_t> logSubsystem, s
     bool isValidLogType = logType == OS_LOG_TYPE_DEFAULT || logType == OS_LOG_TYPE_INFO || logType == OS_LOG_TYPE_DEBUG || logType == OS_LOG_TYPE_ERROR || logType == OS_LOG_TYPE_FAULT;
     MESSAGE_CHECK(isNullTerminated(nullTerminatedLogString) && isValidLogType, m_logStreamConnection->connection());
 
-    // os_log_hook on sender side sends a null category and subsystem when logging to OS_LOG_DEFAULT.
-    auto osLog = OSObjectPtr<os_log_t>();
-    if (isNullTerminated(logSubsystem) && isNullTerminated(logCategory)) {
-        auto subsystem = byteCast<char>(logSubsystem.data());
-        auto category = byteCast<char>(logCategory.data());
-        osLog = adoptOSObject(os_log_create(subsystem, category));
-    }
+    bool subsystemIsNullTerminated = isNullTerminated(logSubsystem);
+    String subsystem = makeString(subsystemIsNullTerminated ? logSubsystem.data() : nullptr, subsystemIsNullTerminated ? "."_s : nullptr, "WebContent."_s, m_pid);
```

Comment:
This will break `WTFSignpostHandleIndirectLog`, since it is doing a pointer comparison on the subsystem/category pair which no longer matches the expected subsystem.

Let me see if I can come up with a small patch to fix it.

Reply:
Never mind, I'll just fix this in a follow-up patch.

Reply:
Ah, great catch! I am putting this patch on hold while investigating other potential issues.

Thanks for reviewing!